### DISCUSSION TITLE
typing Json column

### DISCUSSION THREAD

**Discussion by hyusetiawan** [prisma/prisma#3894] [Q&A] [answered]
I have a table that has "kind" which is an enum and a "config" which is a Json column. The kind enum dictates the shape of the config json, currently I am doing it manually by creating a new type and merging it with the base prisma generated typescript like below:
```ts

type ConfigFactory<K extends ObjectFieldKind, C extends JsonObject> = {
	kind: K
	config: C
}
type ObjectConfigFactory<K extends ConfigFactory<any, any>> = ObjectField & K
type ObjectConfigInputFactory<K extends ConfigFactory<any, any>> = ObjectFieldCreateWithoutSchemaInput & K

// step 1. add the config enum and shape here
export type ObjectConfigText = ConfigFactory<'TEXT', undefined>
export type ObjectConfigOption = ConfigFactory<'OPTION', { options: string[] }>
export type ObjectConfigUser = ConfigFactory<'USER', undefined>
export type ObjectConfigTimerange = ConfigFactory<'TIMERANGE', undefined>

// step 2. add for the object itself here
export type ObjectFieldTypedConfig =
	| ObjectConfigFactory<ObjectConfigText>
	| ObjectConfigFactory<ObjectConfigOption>
	| ObjectConfigFactory<ObjectConfigUser>
	| ObjectConfigFactory<ObjectConfigTimerange>

// the rest
export type ObjectFieldCreateWithoutSchemaInputTypedConfig =
	| ObjectConfigInputFactory<ObjectConfigText>
	| ObjectConfigInputFactory<ObjectConfigOption>
	| ObjectConfigInputFactory<ObjectConfigUser>
	| ObjectConfigInputFactory<ObjectConfigTimerange>

```

is there a better way?

**Comment by ryands17** [comment, marked as answer]
Hey @hyusetiawan 👋 
Currently yes, as this would need to be something that would have to implemented in Prisma's types. Also I would suggest following this https://github.com/prisma/prisma/issues/3219 issue as it is related.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
